06_puppy/marksomething/main.go

```diff
@@ -0,0 +1,91 @@
+package main
+
+import "sync"
+
+// Puppy represents a small dog
+type Puppy struct {
+	ID     uint32
+	Breed  string
+	Colour string
+	Value  int
+}
+
+// PuppyStorer stores an object in a key-value store
+type PuppyStorer interface {
+	CreatePuppy(p Puppy)
+	ReadPuppy(k uint32) Puppy
+	UpdatePuppy(k uint32, p Puppy)
+	DeletePuppy(k uint32)
+}
```

Comment:
None of your interface functions return errors, so how does function caller know if anything goes wrong or transaction success? For example, ReadPuppy or DeletePuppy with non-existing ID?

Reply:
Agreed, add some basic error handling please, that will make lab 6 a lot easier too.